[MIG][15.0] edi_oca

i-vyshnevska opened the pull request:
Depends on

- [x] https://github.com/OCA/connector/pull/425

simahawk committed:
Add edi module

etobella committed:
[IMP] edi: define type, record, record create helper

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

etobella committed:
[IMP] edi: backend add generate_output

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

etobella committed:
[IMP] edi: Set Send function

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

etobella committed:
[IMP] edi: Define a processing function for incoming documents

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

simahawk committed:
edi: small fix/imp on models

simahawk committed:
edi: use freezegun for tests

simahawk committed:
edi: exchange type validate backend vs type

simahawk committed:
edi bump 13.0.1.1.0

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
edi bump 13.0.1.2.0

simahawk committed:
edi: lookup components by convention

simahawk committed:
edi: exchange record backend mandatory

simahawk committed:
edi: exchange type backend ondelete=set null

etobella committed:
[IMP] edi: Add mixins

oca-travis committed:
[UPD] Update edi.pot

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.3.0

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.4.0

simahawk committed:
edi.backend: unify component usage get

simahawk committed:
edi.exchange.record: add handy methods to set/get file

simahawk committed:
edi.backend: add cron to handle record state updates

simahawk committed:
Add edi_storage

simahawk committed:
edi: add specific events, improve record msg lookup

simahawk committed:
edi: pass backend to component ctx

simahawk committed:
edi.exchange.record: improve set/get file

simahawk committed:
edi.backend: improve exception handling

simahawk committed:
edi.backend: add docstrings and TODO

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.5.0

simahawk committed:
edi: improve and fix tests

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.6.0

simahawk committed:
edi: sort out components mixins, add input mixins

simahawk committed:
edi.backend: lookup component by direction too

simahawk committed:
edi: test improve fake components, add input base

simahawk committed:
edi: test process w/ fake components

etobella committed:
[IMP] edi: Refactoring views

etobella committed:
[IMP] edi: input processing, finishing tests

simahawk committed:
edi: fix consumer mixin tests

simahawk committed:
edi: add unique identifier

simahawk committed:
edi: add external identifier

simahawk committed:
edi: fix and organize menu items

simahawk committed:
edi: disable not ready cron

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.7.0

simahawk committed:
edi.backend: return none when component not found

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.8.0

etobella committed:
[IMP] edi: Security

simahawk committed:
edi: simplify edi events

Exclude the code from the name because it can be tricky to hook to
and overall it's a too specific event.

simahawk committed:
edi: simply ack handling

ACK is now provided by a specific type of exchange
and relies on the relation between records of given type.

simahawk committed:
edi.backend: simplify component lookup

etobella committed:
[FIX] edi: Missing filter on cron check

simahawk committed:
edi.backend: handle reception of files

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.8.1

simahawk committed:
edi: move event triggers to exchange record

simahawk committed:
edi: move and rename _exchange_notify_record to exc rec

Notify the related record belongs now to the exchange record itself
as it makes more sense. It has been renamed for better meaning
to '_notify_related_record'.

simahawk committed:
edi.backend: rename check method for exchange_process

simahawk committed:
edi: rename test file for process

simahawk committed:
edi: cron tests silent logger

simahawk committed:
edi: cron generate ack record when needed

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.9.0

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.10.0

simahawk committed:
edi: add receive component base mixin

simahawk committed:
edi: add type advanced settings

simahawk committed:
edi: fix manifest description

simahawk committed:
edi: add base lookup for validators

simahawk committed:
edi.backend: fix handle reception of files

simahawk committed:
edi.backend fix _check_input_exchange_sync signature

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.11.0

simahawk committed:
edi: improve validation handling

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.12.0

etobella committed:
[IMP] edi: Improve readme

etobella committed:
[IMP] edi: Allow to manually generate EDI Exchange records

OCA-git-bot committed:
edi 13.0.1.13.0

simahawk committed:
edi.backend: rename generate to match others

All the other main methods start with 'exchange_'.
Make generate respect this convention.

simahawk committed:
edi: add jobs configuration

etobella committed:
[FIX] edi: Rule not correctly created

OCA-git-bot committed:
edi 13.0.1.13.1

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.14.0

etobella committed:
[IMP] edi: Add a has_exchange_record function to search for the specific exchange record related

etobella committed:
[IMP] edi: Keep old field values on the fields_view_get

Without this, we are loosing the specific views of some fields

OCA-git-bot committed:
edi 13.0.1.14.1

simahawk committed:
edi: consumer mixin fix Serialized field usage

simahawk committed:
edi: component ease access to record

simahawk committed:
edi: refactor component lookup

The match is now done based on specific control keys: backend type, exchange type.
This makes components more clean and their behavior more understandable.

OCA-git-bot committed:
edi 13.0.1.15.0

etobella committed:
[CHG] edi: Use more permissive licence: AGPL-> LGPL

simahawk committed:
edi: cron call exchange action w/ jobs

simahawk committed:
edi: enable cron for input records

simahawk committed:
edi: add TODO for cron methods

simahawk committed:
edi: _get_component fallback to backend model

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.15.1

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.16.0

acsonefho committed:
[FIX] edi: fix record property when there is no model to return the record's parent (if set) + unit test

etobella committed:
[FIX] edi: Fix security & filtering

etobella committed:
[FIX] edi_storage: sort key was not working properly

etobella committed:
[IMP] edi: Change security function

OCA-git-bot committed:
edi 13.0.1.17.0

etobella committed:
[IMP] edi_oca: black, isort, prettier

etobella committed:
[MIG] edi_oca: Migration to 14.0

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[FIX] edi_oca: Missing changes of name

OCA-git-bot committed:
edi_oca 14.0.1.0.1

sbidoul committed:
[IMP] update dotfiles

Yvesldff committed:
Added translation using Weblate (French)

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 56.5% (78 of 138 strings)

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/fr/

LoisRForgeFlow committed:
[FIX] edi_oca: show the exchange file name while there is no file yet.

LoisRForgeFlow committed:
[FIX] edi_oca: show the exchange file name while there is no file yet.

simahawk committed:
edi: add _find_existing_exchange_records to backend

LoisRForgeFlow committed:
[FIX] edi: allow the use of listeners.

use `_notify_*` methods when processing a exchange record. This
will allow the use of listeners based on this events.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

OCA-git-bot committed:
edi_oca 14.0.1.2.0

simahawk committed:
edi_backend: update docstring

simahawk committed:
edi_backend: fix var name

simahawk committed:
edi: exc record get file content as b64 optionally

simahawk committed:
edi: improve exchange record UI

simahawk committed:
edi: improve backend view extension

simahawk committed:
edi: allow job channel conf by type

simahawk committed:
edi: specific method to get type settings

simahawk committed:
edi: load main vars at init

simahawk committed:
edi_backend: exclude class name from error txt

simahawk committed:
edi: exclude exception class from error msg

simahawk committed:
edi: improve exchange type and record UI

simahawk committed:
edi: add 'retry' action to exchange record

This action allows to re-schedule the record for the action that failed
depending on its current state and direction.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

etobella committed:
[IMP] edi: Allow to ignore some jobs

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.4.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

simahawk committed:
edi: fix wrong status bar display new exc record

Without this check you get 2 overlapping bars till you select an exc type.

simahawk committed:
edi: fix consumer mixin

Exchange type has no display_name

etobella committed:
[IMP] edi: ext identifier unique by type and backend

It might be possible to have the same external_identifier for different backends and types.

OCA-git-bot committed:
edi_oca 14.0.1.5.0

OCA-git-bot committed:
edi_oca 14.0.1.5.1

simahawk committed:
edi: backend._get_component inject custom ctx

You can now provide additional env context from type settings.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.6.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

OCA-git-bot committed:
edi_oca 14.0.1.6.1

simahawk committed:
edi_oca: remove filename from status msg

The filename is redundant there.
It's pointless to see it every time in the chatter.
If you want to see the details, you go to the exchange record.

simahawk committed:
edi: chatter mgs tmpl fix link

Linking via 'oe' attrs seems not to work.
Having a real URL also allows to open the record in a new tab.

simahawk committed:
edi: exchange record state use badge widget in tree view

This makes the state more prominent and clear.

etobella committed:
[IMP] edi_oca: Pass development status to Beta

OCA-git-bot committed:
[UPD] README.rst

i-vyshnevska committed:
[MIG][15.0] edi_oca

i-vyshnevska commented on edi_oca/__manifest__.py:
```diff
@@ -0,0 +1,46 @@
+    """,
+    "version": "15.0.1.0.0",
+    "website": "https://github.com/OCA/edi",
+    "development_status": "Alpha",
```
@simahawk  maybe Beta

simahawk replied:
it has been done here https://github.com/OCA/edi/pull/488

simahawk replied:
pls include this change

simahawk commented on test_requirements.txt:
```diff
@@ -1,2 +1,2 @@
-
 odoo_test_helper
+odoo-addon-component_event @ git+https://github.com/OCA/connector@refs/pull/425/head#subdirectory=setup/component_event
```
you need connector too

simahawk commented on edi_oca/tests/test_consumer_mixin.py:
```diff
@@ -59,7 +60,7 @@ def _setup_records(cls):
 
     @classmethod
     def tearDownClass(cls):
-        cls.loader.restore_registry()
```
:fire: this is not the same thing. Look at line 29. We still need this.

simahawk replied:
why this has been marked as resolved? As said, this line must stay otherwise FakeModelLoader won't clean up fake models.

i-vyshnevska replied:
maybe diff was shown wrongly but this line was restored to PR, that is why marked as resolved, anyway, it is there now

simahawk commented on edi_oca/tests/test_consumer_mixin.py:
```diff
@@ -10,13 +10,14 @@
 # If you still want to run `edi` tests w/ pytest when this happens, set this env var.
 @unittest.skipIf(os.getenv("SKIP_EDI_CONSUMER_CASE"), "Consumer test case disabled.")
-class TestConsumerMixinCase(EDIBackendCommonTestCase):
+class TestConsumerMixinCase(EDIBackendCommonComponentRegistryTestCase):
```
rollback, this is not the same thing

simahawk commented on edi_oca/tests/test_security.py:
```diff
@@ -6,10 +6,11 @@
 
 
-class TestEDIExchangeRecordSecurity(EDIBackendCommonTestCase):
+class TestEDIExchangeRecordSecurity(EDIBackendCommonComponentRegistryTestCase):
```
rollback

simahawk commented on edi_oca/tests/test_security.py:
```diff
@@ -59,7 +60,7 @@ def _setup_records(cls):
 
     @classmethod
     def tearDownClass(cls):
-        cls.loader.restore_registry()
```
rollback

simahawk commented on test_requirements.txt:
```diff
@@ -0,0 +1,3 @@
+odoo_test_helper
```
wrong file name too. Is `test-requirements.txt`

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.7.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

simahawk commented:
depending PR got merged

i-vyshnevska committed:
[IMP] change development status base_edi

simahawk commented:
@i-vyshnevska for the dev status version update pls pick the commit from the PR I linked otherwise we'll have false positives when fwd porting changes.

simahawk commented:
@i-vyshnevska after this commmit https://github.com/OCA/edi/pull/490/commits/4146f3007bc6f452c36e4fe15d6d65e1cc062092 pls change the module version to 15.0.1.1.0 w/ commit msg `base_edi 15.0.1.1.0` because we'll merge w/ nobump here.

i-vyshnevska committed:
base_edi 15.0.1.1.0

LoisRForgeFlow commented:
@i-vyshnevska could you include this fix https://github.com/OCA/edi/pull/524?

etobella commented:
/ocabot migration edi_oca

etobella commented:
@i-vyshnevska Can you check the conflicts please?

etobella commented:
@i-vyshnevska Can you check the conflicts please?

etobella commented:
@i-vyshnevska tests are not being executed: see https://github.com/OCA/edi/runs/4962974627?check_suite_focus=true#step:8:160

It seems that test-requirements.txt files is renamed as test_requirements.txt. Can you change it? Do you need some help?

simahawk commented:
#530
